src/Swarrot/Processor/MaxExecutionTime/MaxExecutionTimeProcessor.php

```diff
@@ -61,11 +61,7 @@ public function sleep(array $options): bool
      */
     public function process(Message $message, array $options): bool
     {
-        if (true === $this->isTimeExceeded($options)) {
-            return false;
-        }
```

Comment:
At first, I had not planned to remove this check, but after reading "The child processor should be called even if the time is exceeded. " from https://github.com/swarrot/swarrot/pull/227#issuecomment-758773279, I think it's better to remove it, because that check would only ever kick in if Swarrot itself was consuming the time, making redeliveries even harder to troubleshoot.

Reply:
IMO, as soon as a message have been fetched from the broker, but must be processed.
The current implementation is not OK as this processor it will stop the swarrot consumer if the max execution time have been reached **even if the current message have not been consumed yet**.

That's why, the correct implementation should be:
1. process the message
2. check if time is exceeded
3. If it is, return false, otherwise, return the inner processor result

The check must not be removed, it must be moved **after** processing the current message and this is exactly what your current implementation do. :ok_hand:

Reply:
It definitely sounds like the best solution 👍